firefox-ios/firefox-ios-tests/Tests/ClientTests/Toolbar/ToolbarStateTests.swift

```diff
@@ -0,0 +1,568 @@
+        let searchEngineModel = SearchEngineModel(
+            name: "Google",
+            image: UIImage(named: StandardImageIdentifiers.ExtraSmall.chevronDown)!)
+
+        let newState = reducer(
+            initialState,
+            SearchEngineSelectionAction(
+                windowUUID: self.windowUUID,
+                actionType: SearchEngineSelectionActionType.didTapSearchEngine,
+                selectedSearchEngine: searchEngineModel
+            )
+        )
+
+        XCTAssertNotEqual(newState.addressToolbar, initialState.addressToolbar)
+        XCTAssertEqual(newState.navigationToolbar, initialState.navigationToolbar)
+    }
+
+    func test_didClearAlternativeSearchEngineAction_returnsExpectedState() {
+        setupStore()
+        let initialState = createSubject()
+        let reducer = toolbarReducer()
+
+        let newState = reducer(
+            initialState,
+            SearchEngineSelectionAction(
+                windowUUID: self.windowUUID,
+                actionType: SearchEngineSelectionMiddlewareActionType.didClearAlternativeSearchEngine
+            )
+        )
+
+        XCTAssertNil(newState.addressToolbar.alternativeSearchEngine)
+        XCTAssertEqual(newState.navigationToolbar, initialState.navigationToolbar)
+    }
+
+    // MARK: - Private
+    private func createSubject() -> ToolbarState {
+        return ToolbarState(windowUUID: windowUUID)
+    }
+
+    private func toolbarReducer() -> Reducer<ToolbarState> {
+        return ToolbarState.reducer
+    }
+
+    private func loadWebsiteAction(state: ToolbarState, reducer: Reducer<ToolbarState>) -> ToolbarState {
+        return reducer(
+            state,
+            ToolbarAction(
+                url: URL(string: "http://mozilla.com", invalidCharacters: false),
+                isPrivate: false,
+                isShowingNavigationToolbar: true,
+                canGoBack: true,
+                canGoForward: false,
+                lockIconImageName: StandardImageIdentifiers.Large.lockFill,
+                safeListedURLImageName: nil,
+                windowUUID: windowUUID,
+                actionType: ToolbarActionType.urlDidChange
+            )
+        )
+    }
+
+    // MARK: StoreTestUtility
+    func setupAppState() -> AppState {
+        return AppState(
+            activeScreens: ActiveScreensState(
+                screens: [
+                    .browserViewController(
+                        BrowserViewControllerState(
+                            windowUUID: windowUUID
+                        )
+                    ),
+                    .toolbar(
+                        ToolbarState(
+                            windowUUID: windowUUID
+                        )
+                    )
+                ]
+            )
+        )
+    }
+
+    func setupStore() {
+        StoreTestUtilityHelper.setupStore(
+            with: setupAppState(),
+            middlewares: [ToolbarMiddleware().toolbarProvider]
+        )
```

Comment:
Same as the other state tests, can simplify this with a mock store (then no need to instantiate the real middleware):
```suggestion
        mockStore = MockStoreForMiddleware(state: setupAppState())
        StoreTestUtilityHelper.setupStore(with: mockStore)
```

And `var mockStore: MockStoreForMiddleware<AppState>!` class member.